CleverTapSDK/EventDatabase/CTEventDatabase.m

```diff
@@ -0,0 +1,564 @@
+//
+//  CTEventDatabase.m
+//  CleverTapSDK
+//
+//  Created by Nishant Kumar on 25/10/24.
+//  Copyright © 2024 CleverTap. All rights reserved.
+//
+
+#import "CTEventDatabase.h"
+#import "CTConstants.h"
+
+@interface CTEventDatabase()
+
+@property (nonatomic, strong) CleverTapInstanceConfig *config;
+
+@end
+
+@implementation CTEventDatabase {
+    sqlite3 *_eventDatabase;
+    dispatch_queue_t _databaseQueue;
+}
+
++ (instancetype)sharedInstanceWithConfig:(CleverTapInstanceConfig *)config {
```

Comment:
The CTEventDatabase shared instance with config is bit misleading that it can return a different instance based on the config and use a shared instance with same configs. Do we want separate databases per account id?
The CTEventDatabase is only used in the local store, does it need to be a singleton?

Reply:
Removed config part and still used singleton as local store init can be called from other CT instance also. We are not using separate database per account id here. Any suggestion if we want to use separate database per account id?